### DISCUSSION TITLE
suggestions about template syntax

### DISCUSSION THREAD

**Discussion by ShenTengTu** [vlang/v#11845] [Ideas]
Built-in template compilation is a great feature, but I think it can be improved further.

The following suggestions are based on the actual situation I currently use.

Template directives will cause `undefined ident` error at compile time in the following situations:

1. Declares CSS rules prefixed with `@` symbol in the `style` tag：
```html
<style>
@media (max-height: 480px) {
    header {
        height: 100vh;
    }
}
</style>
```

2. URL with `@` symbol:
```html
<a href="https://unsplash.com/@shentengtu">Unsplash profile</a>
```

Currently, template variable declaration does not force the variable namespace to place in curly brackets(`@{my_var}`), so that the compiler will treat `@media` and `@shentengtu` as undefined template variables and cause the compilation error.

At present, it is possible to use  `@@` or `$$` symbols  to escape into normal text，but this is not intuitive and conflicts in semantics (for CSS at least).

I think it is necessary to force the use of curly brackets for template variable declaration.

As for other template directives (like `@if`), they can use the double `@` symbol as prefix.

**Comment by fnetX** [comment]
I'd even prefer not to use the @ at all, but stick to the template common for Go and templating engines written with their ecosystem in mind, that is curly brackets for everything.

**Comment by Terisback** [comment]
@ShenTengTu @@ is already a thing

**Comment by ShenTengTu** [comment]
@Terisback ya, `@@` is for escaping (and other uses ?) 

Without changing the template directive, I think forcing curly bracket for variable declarations (`@{my_var}`) is still necessary.

Other template system implementation using the syntax of
- `{{...}}`
- `{%...%}`
- `<%...%>`
- `${...}`
- `{>...}`
are not easy to conflict.
